sklearn/feature_selection/tests/test_chi2.py

```diff
@@ -49,18 +50,20 @@ def test_chi2():
     assert_array_almost_equal(Xtrans, Xtrans2)
 
 
-def test_chi2_coo():
+@pytest.mark.parametrize("coo_container", COO_CONTAINERS)
+def test_chi2_coo(coo_container):
     # Check that chi2 works with a COO matrix
     # (as returned by CountVectorizer, DictVectorizer)
-    Xcoo = coo_matrix(X)
+    Xcoo = coo_container(X)
     mkchi2(k=2).fit_transform(Xcoo, y)
     # if we got here without an exception, we're safe
 
 
-def test_chi2_negative():
+@pytest.mark.parametrize("csr_container", COO_CONTAINERS)
```

Comment:
Shouldn't this be based on CSR_CONTAINERS as the test uses csr_matrix internally?
```suggestion
@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
```